### PR TITLE
Apply dates_active decorator to midijob and ges_pflegev

### PR DESCRIPTION
**Pull request by hmgaudecker**
### What problem do you want to solve?

Midijobs were introduced in 2004. There used to be a hacky solution setting the gleitzone to have length zero. With the `dates_active` machinery, we can make this nicer.

Similar comments apply to `ges_pflegev`, which was introduced in 1995 and changed its rules in 2005.

### CODE REVIEW

**Comment by hmgaudecker** [comment]
Implements a bunch of things from #558.

File: src/_gettsim/social_insurance_contributions/ges_pflegev.py [hunk 1]
```diff
@@ -247,13 +372,45 @@ def ges_pflegev_beitr_selbst_m(
...
+        See :func:`_ges_krankenv_bemessungsgrundlage_rente_m`.
+    ges_pflegev_beitr_satz
+        See :func:`ges_pflegev_beitr_satz`.
+
+    Returns
+    -------
+    Monthly health insurance contributions for pension income.
+
+    """
+    out = _ges_krankenv_bemessungsgrundlage_rente_m * (ges_pflegev_beitr_satz * 2)
```
Comment (MImmesberger) [inline src/_gettsim/social_insurance_contributions/ges_pflegev.py:398]: I read [here](https://www.spiegel.de/wirtschaft/urteil-rentner-muessen-pflegeversicherung-allein-bezahlen-a-451421.html#) that until March 2004, the public pension fund did pay half of the Pflegeversicherungsbeitrag. I think this is not implemented here.

File: src/_gettsim/social_insurance_contributions/ges_krankenv.py [hunk 2]
```diff
@@ -1,7 +1,48 @@
 from _gettsim.shared import dates_active
 
 
-def ges_krankenv_beitr_m(  # noqa: PLR0913
+@dates_active(end="2003-03-31", change_name="ges_krankenv_beitr_m")
+def ges_krankenv_beitr_m_vor_midijob(
```
[the remark this answers is not in the document]
Comment (hmgaudecker) [reply src/_gettsim/social_insurance_contributions/ges_krankenv.py:5]: Done.

File: src/_gettsim/social_insurance_contributions/ges_pflegev.py (hunk 1 above)
Comment (hmgaudecker) [reply src/_gettsim/social_insurance_contributions/ges_pflegev.py:398]: Excellent catch! Would prefer to do that separately, though. Can you open an issue and maybe a PR?

**Review by MImmesberger — APPROVED** [review]

**Pull request merged**